linux_cross.nix

```diff
@@ -0,0 +1,69 @@
+let
+  # Here we try to figure out which qemu to use based on the host platform.
+  # This guess can be overriden by passing qemuSuffix
+  qemuByHostPlatform = hostPlatform:
+    # I'd prefer this was a dictionary lookup, with a fall through into abort,
+    # that would make this more readable I guess.  I think there is some similar
+    # mapping somewhere in haskell.nix
+    if hostPlatform.isAarch32
+    then "arm"
+    else if hostPlatform.isAarch64
+    then "aarch64"
+    else abort "Don't know which QEMU to use for hostPlatform ${hostPlatform.config}. Please provide qemuSuffix";
+in
+{ stdenv
+, lib
+, writeScriptBin
+, qemu
+, qemuSuffix ? (qemuByHostPlatform hostPlatform)
+, iserv-proxy
+, remote-iserv
+, gmp
+, extra-test-libs ? []
+, buildPlatform
+, hostPlatform
+, ...
+}:
+let
+  # we want this to hold only for arm (32 and 64bit) for now.
+  isLinuxCross = buildPlatform != hostPlatform && hostPlatform.isLinux && (hostPlatform.isAarch32 || hostPlatform.isAarch64);
+  qemuIservWrapper = writeScriptBin "iserv-wrapper" ''
+    #!${stdenv.shell}
+    set -euo pipefail
+    # Unset configure flags as configure should have run already
+    unset configureFlags
+    PORT=$((5000 + $RANDOM % 5000))
+    (>&2 echo "---> Starting remote-iserv on port $PORT")
+    ${qemu}/bin/qemu-${qemuSuffix} ${remote-iserv}/bin/remote-iserv tmp $PORT &
```

Comment:
You can also use ```${hostPlatform.emulator pkgs}``` to get a qemu-like emulator.

Reply:
@matthewbauer, what’s a qemu-like emulator?

Reply:
@matthewbauer do you mean that `hostPlatform.emulator` will pick out the appropriate emulator for the host platform from the given package set?

Reply:
(A bit of playing around with the repl shows that this is indeed the case, and for the case hostPlatform == buildPlatform, it just gives back a path to a bash shell)

Reply:
@eamsden the only thing I'm concerned about here is that `${hostPlatform.emulator pkgs}` is a very recent addition :-/ It won't work with 18.09 as far as I can see. I'd prefer not to be too aggressive with requiring newer nixpkgs channels.

Reply:
angerman: ok well this is not a priority right now anyway. I'd like to see a successful Aarch32 build before I start making things pretty :)